pipeline/inferrer/feature_extraction/training/src/aws.py

```diff
@@ -1,21 +1,31 @@
 import boto3
 
 
-def get_s3_client(session=None, profile_name=None):
-    if profile_name:
-        session = boto3.session.Session(
-            profile_name=profile_name, region_name="eu-west-1"
-        )
-        s3_client = session.client("s3")
-    elif session:
-        s3_client = session.client("s3")
-    else:
-        raise ValueError(
-            "Need an existing session or a profile name with which to create one"
-        )
+def get_s3_client():
+    try:
+        s3_client = boto3.client("s3")
+    except ClientError as e:
+        logger.error(f"Failed to create s3 client: {e}")
+        raise e
     return s3_client
 
 
-def put_object_to_s3(binary_object, key, bucket_name, session=None, profile_name=None):
-    s3_client = get_s3_client(session, profile_name)
+def put_object_to_s3(binary_object, key, bucket_name):
+    s3_client = get_s3_client()
+    print("Uploading object to S3...")
     s3_client.put_object(Bucket=bucket_name, Key=key, Body=binary_object)
+    print("Uploaded object to S3.")
+
+
+def put_ssm_parameter(path, value, description):
```

Comment:
What are we using SSM for?

Reply:
Ignore me, had to read to the end. Could we not just use S3 to also store the path though - reduce the types of data store we're using.

Reply:
hmmm, we could, but the intention here is for Terraform to be able to get it and inject it as an environment variable.

The work to get the training stuff deployed still [needs to be done](https://github.com/wellcomecollection/platform/issues/4406) and I think will affect this sort of thing, so I might prefer to leave working out the optimal way to get this info until that piece of work.

Reply:
oh, sorry - i imagined this happening at runtime, what you suggest makes sense.